content/articles/how-sweet-is-kubernetes/index.md

```diff
@@ -0,0 +1,267 @@
+- `The Position Simulation` only simulates the positions of the truck at any point in time, and that is its only job.
+Once it reads the position, it hands over the data to the `Message Queue`. The essence of a queue in any architecture of microservice is to avoid loss of data. The queue receives the data, keeping it waiting for the request from the position tracker. The queue implemented here is **ActiveM.** It is also utilized as a docker container. The position tracker gets the information from the queue and performs many calculations, like calculating the speed a particular truck has covered. **API Gateway** is the single entry point to the entire application. Its work is to match all requests incoming to underlying microservices.
+
+![Fleetman microservice](/engineering-education/qoyum_yusuf/fleetman.png)
+
+### What Kubernetes is.
+In proper definition: Kubernetes is an open-source system for automating deployment, scaling, and management of containerized applications.
+
+### What Kubernetes is not.
+Kubernetes is not an alternative to Docker. It is something more.
+
+#### Kubernetes Vs. Docker**
+| s/n | Docker  |  Kubernetes  |
+|----|----------------------------|:----------------------------------    |
+| 1.|Docker is used for building and containerizing images.  |Those containers built, the function of Kubernetes is to manage them effectively; hence you can say Kubernetes is an orchestration tool.  |
+| 2. |Docker runs on a single node. | Kubernetes runs across clusters. |
+| 3. |Docker is built by three y-connector graduates. | Kubernetes is an open-source project built by Google engineers |
+
+However, it is better to see how it works with the aid of the Fleetman application that I deployed on my repo as the instructional material.
+
+### Major Concept in Kubernetes.
+
+#### Deployment
+Deployment is a configuration that provides declarative updates to applications. It allows you to describe an application's life cycle, such as which images to use for the app, the number of pods there should be, and how they should be updated. It is also a configuration where you can determine the number of replica sets to be produced per pod. It helps maintain the cluster in order to avoid the minutest downtime. We can say it is like a blueprint that dictates how every pod feature will behave.
+The configuration we are declaring is in the deployment, while the name is the position simulator. The container we used is on line 17, and the position simulator release `one` from Richard Chesterwood repository. As we can see, 'replicas' is being specified as our replica set, and the template declares our pod. For a standard, `YAML` file even though it can be separated, to ensure orderliness and separation of concerns, we write the services of each deployment in the same file, separating them with three dashes(-). Some do arrange theirs as all services in a separate file. That is also acceptable.
+
+```yaml
```

Comment:
This is also how we write the code blocks as I was trying to explain.
In our instance, we are using yaml so we will label all the codeblocks as yaml.

Reply:
Wow nice